algo/src/main/java/org/neo4j/graphalgo/MSTPrimProc.java

```diff
@@ -43,36 +46,48 @@
 
         ProcedureConfiguration configuration = ProcedureConfiguration.create(config);
 
-        LazyIdMapper idMapper = new LazyIdMapper();
+        LazyIdMapper idMapper = LazyIdMapper.importer(api)
+                .withWeightsFromProperty(propertyName, 1.0)
+                .withOptionalLabel(configuration.getNodeLabelOrQuery())
+                .withOptionalRelationshipType(configuration.getRelationshipOrQuery())
+                .build();
 
         MSTPrimResult.Builder builder = MSTPrimResult.builder();
 
-        ProgressTimer timer = ProgressTimer.start(builder::withLoadDuration);
-        BufferedWeightMap weightMap = BufferedWeightMap.importer(api)
+        CompletableFuture<BufferedWeightMap> weightMap = BufferedWeightMap.importer(api)
```

Comment:
Yea sure. The import logic of dedicated data sources is far from beeing complete. But it was obvious that we can do the imports of several data sources in parallel. Up to now the logic is implemented in the abstract Importer class. But since the MST procedure is the only one which uses the dedicated api2 sources we should defer this to later.